src/test/java/uk/gov/pay/ledger/transaction/TransactionViewDaoITest.java

```diff
@@ -0,0 +1,377 @@
+            aTransactionFixture()
+                    .withGatewayAccountId(gatewayAccountId)
+                    .withCreatedDate(ZonedDateTime.now().minusDays(i))
+                    .insert(rule.getJdbi());
+        }
+
+        TransactionSearchParams searchParams = new TransactionSearchParams();
+        searchParams.setAccountId(gatewayAccountId);
+        searchParams.setFromDate(ZonedDateTime.now().minusDays(1).minusMinutes(10));
+
+        List<TransactionView> viewList = transactionViewDao.searchTransactionView(searchParams);
+
+        assertThat(viewList.size(), Matchers.is(1));
+    }
+
+    @Test
+    public void shouldReturn2Records_withToDateSet() {
+        String gatewayAccountId = "account-id-" + nextLong();
+
+        for (int i = 1; i < 4; i++) {
+            aTransactionFixture()
+                    .withGatewayAccountId(gatewayAccountId)
+                    .withCreatedDate(ZonedDateTime.now().minusDays(i))
+                    .insert(rule.getJdbi());
+        }
+
+        TransactionSearchParams searchParams = new TransactionSearchParams();
+        searchParams.setAccountId(gatewayAccountId);
+        searchParams.setToDate(ZonedDateTime.now().minusDays(2).plusMinutes(10));
+
+        List<TransactionView> viewList = transactionViewDao.searchTransactionView(searchParams);
+
+        assertThat(viewList.size(), Matchers.is(2));
+    }
+
+    @Test
+    public void shouldReturn10Records_withPagesize10() {
+        String gatewayAccountId = "account-id-" + nextLong();
+
+        for (int i = 1; i < 20; i++) {
+            aTransactionFixture()
+                    .withGatewayAccountId(gatewayAccountId)
+                    .insert(rule.getJdbi());
+        }
+
+        TransactionSearchParams searchParams = new TransactionSearchParams();
+        searchParams.setAccountId(gatewayAccountId);
+        searchParams.setDisplaySize(10l);
+
+        List<TransactionView> viewList = transactionViewDao.searchTransactionView(searchParams);
+
+        assertThat(viewList.size(), Matchers.is(10));
+    }
+
+    @Test
+    public void shouldReturn2Records_withOffsetAndPagesizeSet() {
+        String gatewayAccountId = "account-id-" + nextLong();
+        long id = nextLong();
+
+        for (int i = 1; i < 20; i++) {
+            aTransactionFixture()
+                    .withId(id + i)
+                    .withGatewayAccountId(gatewayAccountId)
+                    .withReference("reference" + i)
+                    .insert(rule.getJdbi());
+        }
+
+        TransactionSearchParams searchParams = new TransactionSearchParams();
+        searchParams.setAccountId(gatewayAccountId);
+        searchParams.setDisplaySize(2l);
+        searchParams.setPageNumber(3l);
+
+
+        List<TransactionView> viewList = transactionViewDao.searchTransactionView(searchParams);
+
+        assertThat(viewList.size(), Matchers.is(2));
+        assertThat(viewList.get(0).getReference(), is("reference15"));
```

Comment:
why is this reference 15 on page 3 with page size of 2? I'm probably just being dumb, but can't see how that has worked.

Reply:
This is because search orders results by ID `desc`.  Transaction IDs will start from xxx19 to xxx1.

So for display size of 2
     page 1 should return xxx19, xxx18
     page 2 should return xxx17, xxx16
     page 3 should return xxx15, xxx14

Reply:
I see